pkg/client/linstor.go

```diff
@@ -1124,8 +1125,15 @@ func (s *Linstor) setDevReadWrite(srcPath string) error {
 }
 
 // IsNotMountPoint determines if a directory is a mountpoint.
+//
+// Non-existent paths return (true, nil).
 func (s *Linstor) IsNotMountPoint(target string) (bool, error) {
-	return s.mounter.IsNotMountPoint(target)
+	notMounted, err := s.mounter.IsNotMountPoint(target)
+	if err != nil && !os.IsNotExist(err) {
+		return false, err
+	}
+
+	return notMounted, nil
```

Comment:
So there is the case where it was an `err`, and where it was an `os.IsNotExist()`, then you return what `IsNotMountPoint` gave you for `notMounted`. Even if that works in this particular implementation, the common rule/"contract" IMO is: if it es an `error`, you can not rely on the other return values what so ever, it is allowed to be anything.

Please rephrase that.

Reply:
You are absolutely right on that, updated it just now.